cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/messaging/aws/SnsTopic.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2014 Click Travel Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.clicktravel.infrastructure.messaging.aws;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.auth.policy.Policy;
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.SetTopicAttributesRequest;
+import com.amazonaws.services.sns.model.SubscribeRequest;
+import com.clicktravel.cheddar.infrastructure.messaging.exception.MessagePublishException;
+
+/**
+ * A representative for an actual existing AWS SNS topic. Provides some convenience methods for working with the actual
+ * AWS SNS topic.
+ */
+public class SnsTopic {
```

Comment:
No, the class SnsExchange<T> implements Exchange<T>.
SnsTopic is a supporting class which represents the underlying AWS resource used to implement SnsExchange<T>.